src/libraries/Common/src/System/Security/Cryptography/SLHDsaAlgorithm.cs

```diff
@@ -20,112 +22,215 @@ internal sealed class SLHDsaAlgorithm
         public string Name { get; }
 
         /// <summary>
-        ///  Initializes a new instance of the <see cref="SLHDsaAlgorithm" /> structure with a custom name.
+        ///  Gets the size of the secret key in bytes for this algorithm.
+        /// </summary>
+        /// <value>
+        ///  The size of the secret key in bytes for this algorithm.
+        /// </value>
+        public int SecretKeySizeInBytes { get; }
+
+        /// <summary>
+        ///  Gets the size of the public key in bytes for this algorithm.
+        /// </summary>
+        /// <value>
+        ///  The size of the public key in bytes for this algorithm.
+        /// </value>
+        public int PublicKeySizeInBytes { get; }
+
+        /// <summary>
+        ///  Gets the size of the signature in bytes for this algorithm.
+        /// </summary>
+        /// <value>
+        ///  The size of the signature in bytes for this algorithm.
+        /// </value>
+        public int SignatureSizeInBytes { get; }
+        public string Oid { get; }
+
+        /// <summary>
+        ///  Gets the size of the private seed in bytes for this algorithm.
+        /// </summary>
+        /// <value>
+        ///  The size of the private seed in bytes for this algorithm.
+        /// </value>
+        internal int PrivateSeedSizeInBytes { get; }
+
+        /// <summary>
+        ///  Initializes a new instance of the <see cref="SlhDsaAlgorithm" /> structure with a custom name.
         /// </summary>
         /// <param name="name">
         ///   The name of the algorithm.
         /// </param>
-        private SLHDsaAlgorithm(string name)
+        /// <param name="n">
+        ///   The "security parameter" as described in FIPS 205.
+        /// </param>
+        /// <param name="signatureSizeInBytes">
+        ///   The size of the signature in bytes for this algorithm.
+        /// </param>
+        /// <param name="oid">
+        ///   The Object Identifier (OID) for this algorithm.
+        /// </param>
+        private SlhDsaAlgorithm(string name, int n, int signatureSizeInBytes, string oid)
         {
             Name = name;
+
+            // The seed, secret key and public key sizes are shown to be n, 4n and 2n respectively in
+            // section 9.1 "Key Generation", particularly figure 15 and 16.
+            PrivateSeedSizeInBytes = n;
+            SecretKeySizeInBytes = 4 * n;
+            PublicKeySizeInBytes = 2 * n;
+            SignatureSizeInBytes = signatureSizeInBytes;
+            Oid = oid;
         }
 
-        // TODO: Our algorithm names generally match CNG.  If they don't in this case, consider changing the values.
+        // Slh-DSA parameter sets, and the sizes associated with them,
```

Comment:
Bad find/replace.  All of the ones with a hyphen should be all caps.  Only the type names should be mixed case.